plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/session/CodeGenerationState.kt

```diff
@@ -31,11 +33,13 @@ class CodeGenerationState(
     override var approach: String,
     val config: SessionStateConfig,
     val uploadId: String,
-    val currentIteration: Int,
+    override var currentIteration: Int? = 0,
     val repositorySize: Double,
     val messenger: MessagePublisher,
-    var codeGenerationRemainingIterationCount: Int? = null,
-    var codeGenerationTotalIterationCount: Int? = null,
+    override var codeGenerationRemainingIterationCount: Int? = null,
+    override var codeGenerationTotalIterationCount: Int? = null,
+    var currentCodeGenerationId: String? = "EMPTY_CURRENT_CODE_GENERATION_ID",
+    override var token: CancellationTokenSource?,
```

Comment:
why are these in the constructor? aren't they inherent properties of the state?

Reply:
Its inherit, but also modified by the the current context, so need be overrided.